cluster_utils/grid_search.py

```diff
@@ -1,33 +1,38 @@
+"""Run grid_search over hyperparameters."""
+
 import logging
 import os
 import sys
 from collections import Counter
 from pathlib import Path
 
-from . import grid_search, read_params_from_cmdline
+from . import grid_search
 from .constants import FULL_DF_FILE
 from .git_utils import make_git_params
 from .latex_utils import SectionFromJsonHook, StaticSectionGenerator
-from .settings import GenerateReportSetting, SingularitySettings
+from .settings import (
+    GenerateReportSetting,
+    SingularitySettings,
+    init_main_script_argument_parser,
+    read_main_script_params_from_args,
+)
 from .utils import (
-    check_import_in_fixed_params,
     get_time_string,
     make_temporary_dir,
-    rename_import_promise,
     save_report_data,
 )
 
-if __name__ == "__main__":
-    logger = logging.getLogger("cluster_utils")
 
-    params = read_params_from_cmdline(
-        verbose=False,
-        pre_unpack_hooks=[check_import_in_fixed_params],
-        post_unpack_hooks=[
-            rename_import_promise,
-            GenerateReportSetting.parse_generate_report_setting_hook,
-        ],
-    )
+def grid_search_main() -> int:
```

Comment:
Nit: Any particular reason why this is not just `main`? It's never imported anywhere, right?

Reply:
If I remember correctly, I already had #32 in mind.  But maybe a bit premature, as it isn't even clear yet, how exactly we'll implement this.  I'll rename to just "main".